analysis/src/Symbols.ml

```diff
@@ -134,19 +134,42 @@ let command ~path =
     let parser = Res_driver.parsingEngine.parseInterface ~forPrinter:false in
     let {Res_driver.parsetree = signature} = parser ~filename:path in
     iterator.signature iterator signature |> ignore);
-  let result =
-    !symbols
-    |> List.rev_map (fun (name, loc, kind) ->
-           Protocol.stringifyDocumentSymbolItem
-             {
-               name;
-               location =
-                 {
-                   uri = Uri.toString (Uri.fromPath path);
-                   range = Utils.cmtLocToRange loc;
-                 };
-               kind = kindNumber kind;
-             })
-    |> String.concat ",\n"
-  in
-  print_endline ("[\n" ^ result ^ "\n]")
+  !symbols
+  |> List.rev_map (fun (name, loc, kind) ->
+         let symbol : Protocol.documentSymbolItem =
+           {
+             name;
+             location =
+               {
+                 uri = Uri.toString (Uri.fromPath path);
+                 range = Utils.cmtLocToRange loc;
+               };
+             kind = kindNumber kind;
+           }
+         in
+         symbol)
+
+let workspace ~dir =
+  let open FindFiles in
```

Comment:
There are a bunch of places in the codebase with this kind of code.
Wondering if it's similar enough to grant some refactoring.
Or if the differences are too big to gain something.

E.g. the functionality to read `sourceDirectories` could be factored out.

Reply:
Should we add a separate issue to track this @cristianoc ?

Reply:
I think this can be assessed now, and any follow-on work postponed.

Reply:
@aspeddro will you have a look at this? After that, this should be good for merging.